src/astro/databases/google/bigquery.py

```diff
@@ -36,6 +36,9 @@ def sqlalchemy_engine(self) -> Engine:
     def default_metadata(self) -> Metadata:
         return Metadata(schema=settings.SCHEMA, database=self.hook.project_id)
 
+    def schema_exists(self, schema):
+        return False
```

Comment:
Why are we returning `False` here?
Should we check if the BQ namespace exists?

Reply:
@tatiana in the old `schema_exists` we return False if the DB is not postgres or snowflake 

```def schema_exists(hook, schema, conn_type):
    if conn_type in ["postgresql", "postgres"]:
        created_schemas = hook.run(
            "SELECT schema_name FROM information_schema.schemata;",
            handler=lambda x: [y[0] for y in x.fetchall()],
        )
        return schema.upper() in [c.upper() for c in created_schemas]
    elif conn_type == "snowflake":
        created_schemas = [
            x["SCHEMA_NAME"]
            for x in hook.run("SELECT SCHEMA_NAME from information_schema.schemata;")
        ]
        return schema.upper() in [c.upper() for c in created_schemas]
    return False
```

I was following existing code for parity.

Reply:
Since the schema creation statement has `not exist`, do we need this function to check if a schema exists?
```
_create_schema_statement: str = "CREATE SCHEMA IF NOT EXISTS {}"
```
If we decide we need it, I think it is worth implementing it in all the databases which support schemas.

In the case of BQ, we could use the hook:
https://airflow.apache.org/docs/apache-airflow-providers-google/stable/_api/airflow/providers/google/cloud/hooks/bigquery/index.html#airflow.providers.google.cloud.hooks.bigquery.BigQueryHook.get_dataset

Reply:
@dimberman - do we ever use 'check schema' check without 'create schema'? If not we should merge it. 

And create_schema() can be a good place to raise a Custom exception if we fail to create the schema in the case where the user lack privilege.